Date: 2005-08-02
Initial Package Version: 2.9.6
Upstream Status: http://bugs.archlinux.org/index.php?do=details&id=3116
Origin: Miklos Vajna <vmiklos@frugalware.org>
Description: Adds an option to disable the usage of ccache during build.
 doc/makepkg.8.in |    3 +++
 scripts/makepkg  |   10 ++++++++--
 2 files changed, 11 insertions(+), 2 deletions(-)
diff -Naur pacman-2.9.6.orig/doc/makepkg.8.in pacman-2.9.6/doc/makepkg.8.in
--- pacman-2.9.6.orig/doc/makepkg.8.in	2005-08-02 21:51:06.000000000 +0200
+++ pacman-2.9.6/doc/makepkg.8.in	2005-08-02 21:55:12.000000000 +0200
@@ -368,6 +368,9 @@
 run another copy of makepkg to build and install the missing dependencies.
 The child makepkg calls will be made with the \fB-b\fP and \fB-i\fP options.
 .TP
+.B "\-B, \-\-noccache"
+Do not use ccache during build.
+.TP
 .B "\-c, \-\-clean"
 Clean up leftover work files/directories after a successful build.
 .TP
diff -Naur pacman-2.9.6.orig/scripts/makepkg pacman-2.9.6/scripts/makepkg
--- pacman-2.9.6.orig/scripts/makepkg	2005-08-02 21:51:06.000000000 +0200
+++ pacman-2.9.6/scripts/makepkg	2005-08-02 21:54:28.000000000 +0200
@@ -189,6 +189,7 @@
 	echo "options:"
 	echo "  -a, --searchdeps Search for package dependencies automatically"
 	echo "  -b, --builddeps  Build missing dependencies from source"
+	echo "  -B, --noccache   Do not use ccache during build"
 	echo "  -c, --clean      Clean up work files after build"
 	echo "  -C, --cleancache Clean up source files from the cache"
 	echo "  -d, --nodeps     Skip all dependency checks"
@@ -229,6 +230,7 @@
 DEP_BIN=0
 DEP_SUDO=0
 DEP_SRC=0
+NOCCACHE=0
 NODEPS=0
 FORCE=0
 NOEXTRACT=0
@@ -249,6 +251,7 @@
 		--syncdeps)   DEP_BIN=1 ;;
 		--sudosync)   DEP_SUDO=1 ;;
 		--builddeps)  DEP_SRC=1 ;;
+		--noccache)   NOCCACHE=1 ;;
 		--nodeps)     NODEPS=1 ;;
 		--noextract)  NOEXTRACT=1 ;;
 		--install)    INSTALL=1 ;;
@@ -271,12 +274,13 @@
 			exit 1
 			;;
 		-*)
-			while getopts "acCsSbdehifgGj:l:Lkmnorp:uw:-" opt; do
+			while getopts "acCsSbBdehifgGj:l:Lkmnorp:uw:-" opt; do
 				case $opt in
 					a) SEARCHDEPS=1 ;;
 					c) CLEANUP=1 ;;
 					C) CLEANCACHE=1 ;;
 					b) DEP_SRC=1 ;;
+					B) NOCCACHE=1 ;;
 					d) NODEPS=1 ;;
 					e) NOEXTRACT=1 ;;
 					f) FORCE=1 ;;
@@ -718,7 +722,9 @@
 fi
 
 # use ccache if it's available
-[ -d /usr/lib/ccache/bin ] && export PATH=/usr/lib/ccache/bin:$PATH
+if [ "$NOCCACHE" = "0" ]; then
+	[ -d /usr/lib/ccache/bin ] && export PATH=/usr/lib/ccache/bin:$PATH
+fi
 
 # build
 msg "Starting build()..."
